com.hazelcast.jet.impl.connector.WriteFilePTest.stressTest_exactlyOnce_forceful

_master_ (commit b86829895be06f8152f61d96ca9d96439dd55d24)

Failed on Oracle JDK 11: http://jenkins.hazelcast.com/job/jet-oss-master-oracle-jdk11/269/testReport/com.hazelcast.jet.impl.connector/WriteFilePTest/stressTest_exactlyOnce_forceful/

Stacktrace:
```
java.lang.AssertionError: jobId=04e8-53c0-5708-0001 expected:<RUNNING> but was:<FAILED>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at com.hazelcast.jet.core.JetTestSupport.lambda$assertJobStatusEventually$1(JetTestSupport.java:195)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1356)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1373)
	at com.hazelcast.jet.core.JetTestSupport.assertJobStatusEventually(JetTestSupport.java:194)
	at com.hazelcast.jet.core.JetTestSupport.assertJobStatusEventually(JetTestSupport.java:149)
	at com.hazelcast.jet.impl.connector.WriteFilePTest.stressTest(WriteFilePTest.java:399)
	at com.hazelcast.jet.impl.connector.WriteFilePTest.stressTest_exactlyOnce_forceful(WriteFilePTest.java:365)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Standard output:
```
com.hazelcast.jet.JetException: snapshot with ID 30 doesn't exist or is damaged. Unable to restore the state for job '04e8-53c0-5708-0001', execution 04e8-53c0-571a-0001.
	at com.hazelcast.jet.impl.SnapshotValidator.validateSnapshot(SnapshotValidator.java:49) ~[classes/:?]
	at com.hazelcast.jet.impl.MasterJobContext.rewriteDagWithSnapshotRestore(MasterJobContext.java:359) ~[classes/:?]
	at com.hazelcast.jet.impl.MasterJobContext.lambda$tryStartJob$2(MasterJobContext.java:208) ~[classes/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$42(JobCoordinationService.java:978) ~[classes/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:987) ~[classes/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.submitToCoordinatorThread(JobCoordinationService.java:977) ~[classes/:?]
	at com.hazelcast.jet.impl.MasterJobContext.tryStartJob(MasterJobContext.java:188) ~[classes/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.tryStartJob(JobCoordinationService.java:887) ~[classes/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.restartJob(JobCoordinationService.java:728) ~[classes/:?]
	at com.hazelcast.jet.impl.MasterJobContext.lambda$finalizeJob$17(MasterJobContext.java:629) ~[classes/:?]
	at com.hazelcast.jet.impl.MasterJobContext.lambda$finalizeJob$20(MasterJobContext.java:666) ~[classes/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$42(JobCoordinationService.java:978) ~[classes/:?]
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$43(JobCoordinationService.java:999) ~[classes/:?]
	at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64) [hazelcast-4.0.1.jar:4.0.1]
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217) [hazelcast-4.0.1.jar:4.0.1]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64) [hazelcast-4.0.1.jar:4.0.1]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80) [hazelcast-4.0.1.jar:4.0.1]
```

```
Hiccups measured while running test 'stressTest_exactlyOnce_forceful(com.hazelcast.jet.impl.connector.WriteFilePTest):'
19:38:15, accumulated pauses: 695 ms, max pause: 126 ms, pauses over 1000 ms: 0
19:38:20, accumulated pauses: 1545 ms, max pause: 377 ms, pauses over 1000 ms: 0
19:38:25, accumulated pauses: 687 ms, max pause: 95 ms, pauses over 1000 ms: 0
19:38:30, accumulated pauses: 294 ms, max pause: 10 ms, pauses over 1000 ms: 0
19:38:35, accumulated pauses: 341 ms, max pause: 17 ms, pauses over 1000 ms: 0
19:38:40, accumulated pauses: 490 ms, max pause: 16 ms, pauses over 1000 ms: 0
19:38:45, accumulated pauses: 580 ms, max pause: 24 ms, pauses over 1000 ms: 0
19:38:50, accumulated pauses: 498 ms, max pause: 19 ms, pauses over 1000 ms: 0
19:38:55, accumulated pauses: 456 ms, max pause: 23 ms, pauses over 1000 ms: 0
19:39:00, accumulated pauses: 211 ms, max pause: 10 ms, pauses over 1000 ms: 0
19:39:05, accumulated pauses: 212 ms, max pause: 9 ms, pauses over 1000 ms: 0
19:39:10, accumulated pauses: 136 ms, max pause: 11 ms, pauses over 1000 ms: 0
19:39:15, accumulated pauses: 112 ms, max pause: 11 ms, pauses over 1000 ms: 0
19:39:20, accumulated pauses: 43 ms, max pause: 5 ms, pauses over 1000 ms: 0
19:39:25, accumulated pauses: 57 ms, max pause: 6 ms, pauses over 1000 ms: 0
19:39:30, accumulated pauses: 36 ms, max pause: 2 ms, pauses over 1000 ms: 0
19:39:35, accumulated pauses: 68 ms, max pause: 10 ms, pauses over 1000 ms: 0
19:39:40, accumulated pauses: 51 ms, max pause: 7 ms, pauses over 1000 ms: 0
19:39:45, accumulated pauses: 33 ms, max pause: 1 ms, pauses over 1000 ms: 0
19:39:50, accumulated pauses: 47 ms, max pause: 3 ms, pauses over 1000 ms: 0
19:39:55, accumulated pauses: 33 ms, max pause: 3 ms, pauses over 1000 ms: 0
19:40:00, accumulated pauses: 41 ms, max pause: 10 ms, pauses over 1000 ms: 0
19:40:05, accumulated pauses: 27 ms, max pause: 0 ms, pauses over 1000 ms: 0
19:40:10, accumulated pauses: 29 ms, max pause: 0 ms, pauses over 1000 ms: 0
19:40:15, accumulated pauses: 31 ms, max pause: 1 ms, pauses over 1000 ms: 0
19:40:20, accumulated pauses: 32 ms, max pause: 4 ms, pauses over 1000 ms: 0
19:40:25, accumulated pauses: 146 ms, max pause: 145 ms, pauses over 1000 ms: 0
```

--- a/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/AbstractJobProxy.java
+++ b/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/AbstractJobProxy.java
@@ -71,9 +71,7 @@ public abstract class AbstractJobProxy<T> implements Job {
     }
 
     AbstractJobProxy(T container, long jobId, DAG dag, JobConfig config) {
-        this.jobId = jobId;
-        this.container = container;
-        this.logger = loggingService().getLogger(Job.class);
+        this(container, jobId);
 
         try {
             doSubmitJob(dag, config);

--- a/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/MasterSnapshotContext.java
+++ b/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/MasterSnapshotContext.java
@@ -25,6 +25,7 @@ import com.hazelcast.jet.impl.execution.init.ExecutionPlan;
 import com.hazelcast.jet.impl.operation.SnapshotPhase2Operation;
 import com.hazelcast.jet.impl.operation.SnapshotPhase1Operation;
 import com.hazelcast.jet.impl.operation.SnapshotPhase1Operation.SnapshotPhase1Result;
+import com.hazelcast.jet.impl.util.LoggingUtil;
 import com.hazelcast.logging.ILogger;
 import com.hazelcast.map.IMap;
 import com.hazelcast.spi.impl.operationservice.Operation;
@@ -191,6 +192,12 @@ class MasterSnapshotContext {
             @Nullable CompletableFuture<Void> future
     ) {
         mc.coordinationService().submitToCoordinatorThread(() -> {
+            if (executionId != this.mc.executionId()) {
+                LoggingUtil.logFine(logger, "%s: ignoring responses for snapshot %s phase 1: " +
+                                "the responses are from a different execution: %s. Responses: %s",
+                        mc.jobIdString(), snapshotId, idToString(executionId), responses);
+                return;
+            }
             // Note: this method can be called after finalizeJob() is called or even after new execution started.
             // We only wait for snapshot completion if the job completed with a terminal snapshot and the job
             // was successful.
@@ -283,6 +290,13 @@ class MasterSnapshotContext {
             long startTime
     ) {
         mc.coordinationService().submitToCoordinatorThread(() -> {
+            if (executionId != this.mc.executionId()) {
+                LoggingUtil.logFine(logger, "%s: ignoring responses for snapshot %s phase 2: " +
+                                "the responses are from a different execution: %s. Responses: %s",
+                        mc.jobIdString(), snapshotId, idToString(executionId), responses);
+                return;
+            }
+
             for (Entry<MemberInfo, Object> response : responses) {
                 if (response.getValue() instanceof Throwable) {
                     logger.warning(SnapshotPhase2Operation.class.getSimpleName() + " for snapshot " + snapshotId + " in "

